### PR TITLE
Feature/HOLO-604: implementing critical issue fixes

### PR DESCRIPTION
**Pull request by ACC01ADE**
## Describe Changes

Covers all code4rena audit critical issue (except for 473.

- Re-wrote gas prices and limits to be chain specific
- Fixed issue in HolographERC721 enforcer for re-minting burned NFT
- Updated tests to work with updated code

Code4rena related audit issues:

- https://github.com/code-423n4/2022-10-holograph-findings/issues/505
- https://github.com/code-423n4/2022-10-holograph-findings/issues/497
- https://github.com/code-423n4/2022-10-holograph-findings/issues/445
- https://github.com/code-423n4/2022-10-holograph-findings/issues/364
- https://github.com/code-423n4/2022-10-holograph-findings/issues/353
- https://github.com/code-423n4/2022-10-holograph-findings/issues/339
- https://github.com/code-423n4/2022-10-holograph-findings/issues/331
- https://github.com/code-423n4/2022-10-holograph-findings/issues/297
- https://github.com/code-423n4/2022-10-holograph-findings/issues/294
- https://github.com/code-423n4/2022-10-holograph-findings/issues/290
- https://github.com/code-423n4/2022-10-holograph-findings/issues/288
- https://github.com/code-423n4/2022-10-holograph-findings/issues/261
- https://github.com/code-423n4/2022-10-holograph-findings/issues/247
- https://github.com/code-423n4/2022-10-holograph-findings/issues/244
- https://github.com/code-423n4/2022-10-holograph-findings/issues/241
- https://github.com/code-423n4/2022-10-holograph-findings/issues/223
- https://github.com/code-423n4/2022-10-holograph-findings/issues/218
- https://github.com/code-423n4/2022-10-holograph-findings/issues/183
- https://github.com/code-423n4/2022-10-holograph-findings/issues/178
- https://github.com/code-423n4/2022-10-holograph-findings/issues/177
- https://github.com/code-423n4/2022-10-holograph-findings/issues/176
- https://github.com/code-423n4/2022-10-holograph-findings/issues/171

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] All Hardhat tests are passing

### CODE REVIEW

File: contracts/HolographOperator.sol [hunk 1]
```diff
@@ -1062,6 +1075,26 @@ contract HolographOperator is Admin, Initializable, HolographOperatorInterface {
     }
   }
 
+  /**
+   * @notice Get the Minimum Gas Price
+   * @dev This amount is used as the value that will define a job as underpriced is lower than
```
Comment (alexanderattar) [inline contracts/HolographOperator.sol]: The minimum value required to execute a job without being under priced

File: deploy/07_layer_zero_module.ts [hunk 2]
```diff
@@ -24,20 +25,49 @@ const func: DeployFunction = async function (hre1: HardhatRuntimeEnvironment) {
 
   const salt = hre.deploymentSalt;
 
-  const holograph = await hre.ethers.getContract('Holograph');
+  const MSGBASEGAS: BigNumber = BigNumber.from('110000');
```
Comment (alexanderattar) [inline deploy/07_layer_zero_module.ts]: Nit.. I think adding underscores between words would make these easier to read.

**Review by alexanderattar — APPROVED** [review]
Nice!

File: deploy/07_layer_zero_module.ts (hunk 2 above)
Comment (ACC01ADE) [reply deploy/07_layer_zero_module.ts]: Agreed, will modify.

File: contracts/HolographOperator.sol (hunk 1 above)
Comment (ACC01ADE) [reply contracts/HolographOperator.sol]: Is this a suggested alternative @dev text?

Comment (alexanderattar) [reply contracts/HolographOperator.sol]: yeah i guess so. just found the original one a little unclear

Comment (ACC01ADE) [reply contracts/HolographOperator.sol]: Ok perfect, will update it to this text then.

**Pull request merged**